src/Service/Locale.php

```diff
@@ -22,57 +21,43 @@
 class Locale
 {
     /**
-     * @var State
+     * @var ContentfulLocale[]
      */
-    private $state;
+    private $contentfulLocales = [];
 
     /**
-     * @var Environment
+     * @var ContentfulLocale|null
      */
-    private $environment;
-
-    /**
-     * @var string[]
-     */
-    private $availableLocales;
+    private $current;
 
     /**
      * @param State      $state
      * @param Contentful $contentful
-     * @param string[]   $availableLocales
      */
-    public function __construct(State $state, Contentful $contentful, array $availableLocales)
+    public function __construct(State $state, Contentful $contentful)
     {
-        $this->state = $state;
-        $this->environment = $contentful->findEnvironment();
-        $this->availableLocales = $availableLocales;
+        try {
+            $environment = $contentful->findEnvironment();
+
+            $this->contentfulLocales = $environment->getLocales();
+            $this->current = $environment->getLocale($state->getLocale());
+        } catch (\Exception $exception) {
+        }
```

Comment:
The main problem was here.
It was a dependency chain issue: this class is used by the templating engine, which is used in the `ResponseFactory`, which is used in all the application to generate response objects. This means that all requests end up instantiating this `Locale` class. 

If the current request contains a faulty cookie with invalid credentials, the call to `$contentful->findEnvironment()` would inevitably throw an exception, which would blow up into a `500` error page.

In order to fix that, I simply wrapped the call in a try/catch block, and instead of obtaining the locales later, I pre-fetch them and keep those as object properties.

Calling `$contetnful->findEnvironment()` in certain cir